src/index.ts

```diff
@@ -0,0 +1,543 @@
+   * instance configuration screen.
+   * @return {object[]}
+   */
+  config_fields(): SomeCompanionConfigField[] {
+    return configFields(this);
+  }
+
+  getRestApi(): ArenaRestApi | null {
+    return this._restApi;
+  }
+
+  getOscApi(): ArenaOscApi | null {
+    return this._oscApi;
+  }
+
+  get actions(): CompanionActions {
+    var restApi = this.getRestApi.bind(this);
+    var oscApi = this.getOscApi.bind(this);
+    var actions: CompanionActions = {
+      bypassLayer: bypassLayer(restApi, oscApi),
+      clearAll: clearAllLayers(restApi, oscApi),
+      clearLayer: clearLayer(restApi, oscApi),
+      compNextCol: compNextCol(restApi, oscApi),
+      compPrevCol: compPrevCol(restApi, oscApi),
+      custom: customOscCommand(oscApi),
+      grpNextCol: groupNextCol(restApi, oscApi),
+      grpPrevCol: groupPrevCol(restApi, oscApi),
+      layNextCol: layerNextCol(restApi, oscApi),
+      layPrevCol: layerPrevCol(restApi, oscApi),
+      selectClip: selectClip(restApi, oscApi),
+      soloLayer: soloLayer(restApi, oscApi),
+      tempoTap: tempoTap(restApi, oscApi),
+      triggerClip: connectClip(restApi, oscApi),
+      triggerColumn: triggerColumn(restApi, oscApi),
+    }
+    return actions;
+  }
+
+  setupFeedback() {
+    if (this._restApi) {
+      this.setFeedbackDefinitions({
+        connectedClip: {
+          type: 'boolean',
+          label: 'Connected Clip',
+          style: {
+            color: this.rgb(0, 0, 0),
+            bgcolor: this.rgb(0, 255, 0),
+          },
+          options: [
+            {
+              id: 'layer',
+              type: 'number',
+              label: 'Layer',
+              default: 1,
+              min: 1,
+              max: 65535
+            },
+            {
+              id: 'column',
+              type: 'number',
+              label: 'Column',
+              default: 1,
+              min: 1,
+              max: 65535
+            }
+          ],
+          callback: (feedback: CompanionFeedbackEvent): boolean => {
+            var layer = feedback.options.layer;
+            var column = feedback.options.column;
+            if (layer !== undefined && column !== undefined) {
+              return this.connectedClips.has(`${layer}-${column}`);
+            }
+            return false;
+          },
+          subscribe: (feedback: CompanionFeedbackEvent) => {
+            var layer = feedback.options.layer as number;
+            var column = feedback.options.column as number;
+            if (layer !== undefined && column !== undefined) {
+              this.addClipConnectedSubscription(layer, column);
+            }
+          },
+          unsubscribe: (feedback: CompanionFeedbackEvent) => {
+            var layer = feedback.options.layer as number;
+            var column = feedback.options.column as number;
+            if (layer !== undefined && column !== undefined) {
+              this.addClipConnectedSubscription(layer, column);
```

Comment:
Should this be `addClipConnectedSubscription` or `removeClipConnectedSubscription`?

What happens if:
* You have a button using this feedback with layer: 1, column: 1, so subscribe is called
* You make a copy of the button, subscribe is called for the new button
* You edit that copy to look at layer: 2, column: 2. unsubscribe will be called for layer: 1, column: 1, followed by a subscribe for layer: 2, column: 2
From looking at this, it looks like the subscription for the original feedback will no longer be active, so I guess it will stop updating?

I handled a similar thing in the x32 module by tracking the ids of the feedbacks/actions which were using a subscription. That way, when a call to unsubscribe happens, I can see if there are other places still using the subscription https://github.com/bitfocus/companion-module-behringer-x32/blob/master/src/state.ts#L64-L93

I think this applies to many other feedbacks/actions too